dev_guide/builds/build_inputs.adoc

```diff
@@ -0,0 +1,911 @@
+=== Custom Strategy
+
+When using a `Custom` strategy, all the defined input secrets are available
+inside the builder container in the *_/var/run/secrets/openshift.io/build_*
+directory. The custom build image is responsible for using these secrets
+appropriately. The `Custom` strategy also allows secrets to be defined as
+described in xref:build_strategies.adoc#custom-secrets[Custom Strategy Options].
+
+There is no technical difference between existing strategy secrets and the input
+secrets. However, your builder image might distinguish between them and use them
+differently, based on your build use case.
+
+The inpout secrets are always mounted into the
+*_/var/run/secrets/openshift.io/build_* directory or your builder can parse the
+`$BUILD` environment variable, which includes the full build object.
+
+[[using-external-artifacts]]
+== Using External Artifacts
+
+It is not recommended to store binary files in a source repository. Therefore,
+you may find it necessary to define a build which pulls additional files (such
+as Java *_.jar_* dependencies) during the build process. How this is done
+depends on the build strategy you are using.
+
+For a `Source` build strategy, you must put appropriate shell commands into
+the *_assemble_* script:
+
+.*_.s2i/bin/assemble_* File
+[source,bash]
+----
+#!/bin/sh
+APP_VERSION=1.0
+wget http://repository.example.com/app/app-$APP_VERSION.jar -O app.jar
+----
+
+.*_.s2i/bin/run_* File
+[source,bash]
+----
+#!/bin/sh
+exec java -jar app.jar
+----
+
+[NOTE]
+====
+For more information on how to control which *_assemble_* and *_run_* script is
+used by a Source build, see
+xref:build_strategies.adoc#override-builder-image-scripts[Overriding Builder Image Scripts].
+====
+
+For a `Docker` build strategy, you must modify the *_Dockerfile_* and invoke
+shell commands with the
+link:https://docs.docker.com/engine/reference/builder/#run[`RUN` instruction]:
+
+.Excerpt of *_Dockerfile_*
+----
+FROM jboss/base-jdk:8
+
+ENV APP_VERSION 1.0
+RUN wget http://repository.example.com/app/app-$APP_VERSION.jar -O app.jar
+
+EXPOSE 8080
+CMD [ "java", "-jar", "app.jar" ]
+----
+
+In practice, you may want to use an environment variable for the file location
+so that the specific file to be downloaded can be customized using an
+environment variable defined on the `BuildConfig`, rather than updating the
+*_assemble_* script or *_Dockerfile_*.
+
+You can choose between different methods of defining environment variables:
+
+- xref:build_strategies.adoc#environment-files[Using the *_.s2i/environment_* file] (only for a
+Source build strategy)
+- xref:build_strategies.adoc#buildconfig-environment[Setting in `BuildConfig`]
+- xref:../../cli_reference/basic_cli_operations.adoc#build-and-deployment-cli-operations[Providing
+explicitly using `oc start-build --env`] (only for builds that are triggered
+manually)
+
+
+[[using-docker-credentials-for-private-registries]]
+== Using Docker Credentials for Private Registries
```

Comment:
Plus, given the `[NOTE]` we have about how it's not required for internal registry use, it seemed to make sense to clarify in the section title that this is for private registry use.